### PR TITLE
Web Planning 2016 & OMERO.figure POTN talks

### PR DESCRIPTION
**Pull request by will-moore**
Planning future development directions for webclient, to be presented at group meeting on 19th January, including:
- Overview of existing webclient
- Reasons to modernise / update
- Proposed changes (including React.js)

@aleksandra-tarkowska is also adding a section on backend changes, including caching.

cc @chris-allan Several significant changes to the webclient to be discussed, if you can make the 3.30 pm meeting next Tuesday, or want to discuss at another time?

### CODE REVIEW

**Comment by will-moore** [comment]
Presentation is currently viewable at http://will-moore.github.io/presentations/2016/Web-Planning-Jan-2016/

File: 2016/Web-Planning-Jan-2016/index.html [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+                font-size: 40px;
+            }
+            .ome_theme {
+                background-image: url("images/ome-icon-opacity.png");
+                background-color: #f2f6f9;
+                background-repeat: no-repeat;
+                background-position: right -270px top -120px;
+            }
+            .reveal {
+                text-transform: none;   !important
```
Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html]: it's okay to have an `!important` _after_ the semicolon?

File: 2016/Web-Planning-Jan-2016/index.html [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+                </li>
+
+            </ul>
+        </section>
+
+        <section>
+            <h2>Current state of OMERO.web (...3)</h2>
+            <ul>
+                <li>Approx 5k lines of Python in webclient views.py</li>
+                <li>Not very many RESTfull webservices.</li>
```
Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html]: RESTful (also below)

Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html]: suddenly start ending with periods, most slides don't use that style

File: 2016/Web-Planning-Jan-2016/index.html [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+        ...
+</code></pre>
+        </section>
+
+
+        <section data-background="images/ReactCentrePanel.png" data-background-transition="slide">
+        </section>
+
+        
+        <section data-background="images/ReactPlate.png" data-background-transition="slide">
```
Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html:304]: I'm assuming these react screenshots don't include non-public data.

File: 2016/Web-Planning-Jan-2016/index.html [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+        <section>
+            <h2>Build tools</h2>
+
+            <ul>
+                <li>Concatenate, transpile (JSX, ES2015), minimise, dependency management etc</li>
+                <li>Currently we use Django pipeline (can use Babel for ES2015)</li>
+                <li>Grunt (general purpose task runner, OMERO.figure)</li>
+                <li>webpack (more recent, <a href="https://github.com/MicronOxford/webtagging/pull/37">
+                        OMERO.webtagging #37
+                    </a>
```
Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html]: close parenthesis

File: 2016/Web-Planning-Jan-2016/index.html [hunk 5]
```diff
@@ -0,0 +1,478 @@
...
+                </li>
+                <li>Others (gulp, brocolli etc)</li>
```
Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html]: broccoli

**Comment by mtbc** [comment]
Not blocking this PR but I wonder if, by version, we want to put the reveal.js code/files somewhere common rather than duplicating per presentation.

File: 2016/Web-Planning-Jan-2016/index.html [hunk 6]
```diff
@@ -0,0 +1,478 @@
...
+                <li>Search as a separate web application</li>
+                <li>Useful for public repository</li>
+            </ul>
+        </section>
+
+        <section>
+            <h2>Cache backend <a href="https://trac.openmicroscopy.org/ome/ticket/11517">#11517</a></h2>
+            <ul>
+                <li>Memcached - simple in memory cache.</li>
+                <li>Redis - flexible and complex an in memory key-value store: highly scalable, persistant data store.</li>
```
Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html]: in-memory

Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html]: persistent

**Comment by sbesson** [comment]
@mtbc: the general assumption is that we do not need a proof-reading of these presentations as thorough as the docs. I would focus on the critical elements.

**Comment by mtbc** [comment]
This proof-reading was nowhere near as thorough as for the docs. :smiley:

**Comment by mtbc** [comment]
What "critical elements" do you have in mind? Are things like spelling errors or unmatched parentheses something we should ignore? I'm happy to fit in with whatever the policy is: mostly I was aiming for things that are distracting from the actual content.

**Comment by hflynn** [comment]
Things distracting from the content should definitely be picked up. I'd be inclined to point out spelling errors (actual errors not British English) because if it was my presentation I'd want to fix them before promoting externally. Unmatched parentheses and such are kind of by-the-by I'd have thought, unless they make the content confused.

**Comment by will-moore** [comment]
I've fixed all the small issues @mtbc identified, apart from the screenshots. I don't know that the data is public (probably don't have explicit public permissions), but tried to pick obscure data that was practically black and didn't have revealing names etc.

**Comment by mtbc** [comment]
Yeah, I let the British spelling pass.

File: 2016/Web-Planning-Jan-2016/index.html [hunk 7]
```diff
@@ -0,0 +1,478 @@
...
+                <li>Search as a separate web application</li>
+                <li>Useful for public repository</li>
+            </ul>
+        </section>
+
+        <section>
+            <h2>Cache backend <a href="https://trac.openmicroscopy.org/ome/ticket/11517">#11517</a></h2>
+            <ul>
+                <li>Memcached - simple in memory cache</li>
+                <li>Redis - flexible and complex an in memory key-value store: highly scalable, persistant data store</li>
```
Comment (mtbc) [inline 2016/Web-Planning-Jan-2016/index.html]: still in-memory, persistent

**Comment by will-moore** [comment]
Most of the Dataset thumbnails are either blank/grey or images from the EMDB so they are already public. The Plate is called Plate1 and most of those thumbnails are practically black too, so it's practically impossible to identify it and could hardly be considered a release of private data. So unless anyone objects, I think this is good to merge.

**Comment by will-moore** [comment]
Dundee users' meeting presentation on OMERO.figure added: see http://will-moore.github.io/presentations/2016/Dundee-meeting-OMERO.figure/

**Pull request merged**